File: backend/src/main/java/br/ufpr/sept/so2/modules/bff/application/ports/PeriodoVigenteQueryPort.java
package br.ufpr.sept.so2.modules.bff.application.ports;

import java.time.LocalDate;
import java.util.Optional;
import java.util.UUID;

public interface PeriodoVigenteQueryPort {

    Optional<PeriodoVigenteResumo> consultar(LocalDate data);

    record PeriodoVigenteResumo(UUID id, int ano, int semestre, LocalDate inicio, LocalDate fim) {

        public String rotulo() {
            return ano + "/" + semestre;
        }
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/bff/application/ports/PeriodoVigenteQueryPort.kt
package br.ufpr.sept.so2.modules.bff.application.ports

import java.time.LocalDate
import java.util.UUID

fun interface PeriodoVigenteQueryPort {
    fun consultar(data: LocalDate): PeriodoVigenteResumo?

    data class PeriodoVigenteResumo(
        val id: UUID,
        val ano: Int,
        val semestre: Int,
        val inicio: LocalDate,
        val fim: LocalDate,
    ) {
        fun rotulo(): String = "$ano/$semestre"
    }
}
